src/@batch-flask/core/data/list-getter/list-getter.ts

```diff
@@ -70,38 +70,39 @@ export abstract class ListGetter<TEntity extends Record<any>, TParams> extends G
             return of(cachedResponse);
         }
 
-        return this.list(params, options).pipe(map(x => this._processItems(cache, x, params, options, true)));
+        return this.list(params, options).pipe(flatMap(x => this._processItems(cache, x, params, options, true)));
     }
 
     private _fetchNext(token: ContinuationToken): Observable<ListResponse<TEntity>> {
         const cache = this.getCache(token.params);
-        return this.listNext(token).pipe(map(x => this._processItems(cache, x, token.params, token.options, false)));
+        return this.listNext(token).pipe(flatMap(x => this._processItems(cache, x, token.params, token.options, false)));
     }
 
     private _processItems(
         cache: DataCache<TEntity>,
         response: any,
         params: TParams,
         options: ListOptions,
-        isFirstPage: boolean): ListResponse<TEntity> {
+        isFirstPage: boolean): Observable<ListResponse<TEntity>>{
 
         const { data, nextLink } = response;
-        const items = data.map(x => this._createItem(x, params));
-        const keys = OrderedSet(cache.addItems(items, options.select));
+        const items: TEntity[] = data.map(x => this._createItem(x, params));
+        const keys: OrderedSet<string> = OrderedSet(cache.addItems(items, options.select));
         const token = {
             nextLink,
             params,
             options,
         };
 
-        if (items.size !== 0 && isFirstPage) {
+        if (items.length !== 0 && isFirstPage) {
```

Comment:
are you sure this is correct? Did it change from a Immutable List back to a regular array? Maybe we should add type to response here instead of any

Reply:
Array.map returns an array

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
yeah my question was more about the data  being an array vs a List

Reply:
Ah yea its an array 
![image](https://user-images.githubusercontent.com/18502617/70658377-baf79c00-1c12-11ea-8564-1114cfa194f6.png)